src/mappings/entities/mapSettlement.ts

```diff
@@ -140,7 +139,7 @@ async function handleInstructionUpdate(params: Codec[], event: SubstrateEvent) {
   const instructionId = getTextValue(params[2]);
   const instruction = await Instruction.get(instructionId);
   if (instruction) {
-    instruction.addresses.push(signer);
+    if (signer) instruction.addresses.push(signer);
```

Comment:
This is being done in the handlenstructionFinalized method, so I figure I'd put it here just to be safe.